org.eclipse.jdt.ls.start/src/org/eclipse/jdt/ls/start/JavaLanguageServerStart.java

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.start;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Plugin;
+import org.eclipse.core.runtime.Status;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.SubstituteLoggerFactory;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.joran.JoranConfigurator;
+import ch.qos.logback.core.joran.spi.JoranException;
+import ch.qos.logback.core.util.StatusPrinter;
+
+/**
+ * Override logback preferences *before* M2E plugin is activated below
+ *
+ * @author snjeza
+ *
+ */
+public class JavaLanguageServerStart extends Plugin {
+
+	private static final String LOGBACK_CONFIG_FILE_PROPERTY = "logback.configurationFile";
+	private static final String LOGBACK_DEFAULT_FILENAME = "logback.xml";
+	private static final String LOGBACK_EMPTY_FILENAME = "emptylogback.xml";
+	public static final String PLUGIN_ID = "org.eclipse.jdt.ls.start";
```

Comment:
If we have to create a new bundle to solve this problem, I would rename it from "org.eclispe.jdt.ls.start" to "**org.eclipse.jdt.ls.logback**", and make this bundle only used for handling logback logic.

Reply:
Fixed.